### PR TITLE
[SELC-5806] feat: Refactored class Product

### PR DESCRIPTION
**Pull request by giampieroferrara**
#### List of Changes

Added new attribute institutionContractsMapping, userContractsMapping and alias into product.
Renamed class ContractStorage -> ContractTemplate
Deleted unused method fillContractTemplatePathAndVersion
Into class ContractTemplate deleted attr about date
Added new test case into Product
Upadated SDK

#### Motivation and Context

For better maintainability and code quality

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/entity/OnboardingWorkflow.java [hunk 1]
```diff
@@ -6,47 +6,58 @@
...
-    public OnboardingWorkflow() {
-    }
+  public OnboardingWorkflow() {}
 
-    protected Onboarding onboarding;
-    public abstract String emailRegistrationPath(MailTemplatePathConfig config);
+  protected Onboarding onboarding;
 
-    public abstract String getEmailCompletionPath(MailTemplatePathConfig config);
+  public abstract String emailRegistrationPath(MailTemplatePathConfig config);
```
Comment (andrea-putzu) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/entity/OnboardingWorkflow.java:29]: why this method is not a get?

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingService.java [hunk 2]
```diff
@@ -100,9 +99,10 @@ public void createContract(OnboardingWorkflow onboardingWorkflow) {
         contractService.createContractPDF(onboardingWorkflow.getContractTemplatePath(product), onboarding, manager, delegates, product.getTitle(), onboardingWorkflow.getPdfFormatFilename());
     }
 
+    // TODO
     public void loadContract(Onboarding onboarding) {
         Product product = productService.getProductIsValid(onboarding.getProductId());
-        contractService.loadContractPDF(product.getContractTemplatePath(), onboarding.getId(), product.getTitle());
+        contractService.loadContractPDF(product.getInstitutionContractTemplate(onboarding.getInstitution().getInstitutionType().name()).getContractTemplatePath(), onboarding.getId(), product.getTitle());
```
Comment (andrea-putzu) [inline apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingService.java:105]: could you check if onboarding.getInstitution() is null?

Comment (giampieroferrara) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingService.java]: Fixed!

File: apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/entity/OnboardingWorkflow.java (hunk 1 above)
Comment (giampieroferrara) [reply apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/entity/OnboardingWorkflow.java:28]: Fixed w @pierpaolodidato89 !

**Review by andrea-putzu — APPROVED** [review]

**Pull request merged**